src/wei/core/workflow.py

```diff
@@ -126,6 +126,22 @@ def save_workflow_files(wf_run: WorkflowRun, files: List[UploadFile]) -> Workflo
     return wf_run
 
 
+def pause_workflow_run(wf_run: WorkflowRun) -> None:
+    """Pauses the workflow run"""
+    wf_run.status = WorkflowStatus.PAUSED
+    with state_manager.wc_state_lock():
+        state_manager.set_workflow_run(wf_run)
+    return wf_run
+
+
+def resume_workflow_run(wf_run: WorkflowRun) -> None:
+    """Resumes the workflow run"""
+    wf_run.status = WorkflowStatus.IN_PROGRESS
```

Comment:
I think it makes sense to set the status as IN_PROGRESS here, because the workflow may or may not resume running immediately, depending on the module. But we should have some logic in the scheduler to then update the state to RUNNING if the module does resume.